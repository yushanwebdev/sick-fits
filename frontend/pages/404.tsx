import styled from 'styled-components';

const WrapperStyles = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;

  p:first-child {
    font-size: 3rem;
    font-weight: 500;
    border-right: 1px solid #000;
    padding-right: 20px;
  }

  p:nth-child(2) {
    font-size: 2rem;
    padding-left: 20px;
  }
`;

export default function Custom404() {
  return (
    <WrapperStyles>
      <p>404</p>
      <p>This page could not be found.</p>
    </WrapperStyles>
  );
}
