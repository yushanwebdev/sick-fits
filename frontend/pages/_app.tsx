/* eslint-disable @typescript-eslint/no-unsafe-assignment */
/* eslint-disable @typescript-eslint/ban-ts-comment */
import { ApolloProvider } from '@apollo/client';
import type { AppProps } from 'next/app';
import { Router } from 'next/router';
import styled, { ThemeProvider, DefaultTheme } from 'styled-components';
import NProgress from 'nprogress';
import Header from '../components/Header';
import GlobalStyle from '../components/styles/GlobalStyles';
import { useApollo } from '../lib/apolloClient';

import '../components/styles/nprogress.css';

const theme: DefaultTheme = {
  colors: {
    red: '#ff0000',
    black: '#393939',
    grey: '#3a3a3a',
    get gray() {
      return (this as DefaultTheme['colors']).grey;
    },
    lightGrey: '#e1e1e1',
    get lightGray() {
      return (this as DefaultTheme['colors']).lightGrey;
    },
    offWhite: '#ededed',
    maxWidth: '1000px',
    bs: '0 12px 24px 0 rgba(0, 0, 0, 0.09)',
  },
};

const InnerStyles = styled.div`
  max-width: var(--maxWidth);
  margin: 0 auto;
  padding: 2rem;
`;

Router.events.on('routeChangeStart', () => NProgress.start());
Router.events.on('routeChangeComplete', () => NProgress.done());
Router.events.on('routeChangeError', () => NProgress.done());

const MyApp = ({ Component, pageProps }: AppProps) => {
  const apolloClient = useApollo(pageProps);

  return (
    <ApolloProvider client={apolloClient}>
      <ThemeProvider theme={theme}>
        <GlobalStyle />
        <Header />
        <InnerStyles>
          <Component {...pageProps} />
        </InnerStyles>
      </ThemeProvider>
    </ApolloProvider>
  );
};

export default MyApp;
