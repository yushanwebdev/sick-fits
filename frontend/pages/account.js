export default function Home() {
    return (
      <div>
        <p>Account</p>
      </div>
    )
  }
  