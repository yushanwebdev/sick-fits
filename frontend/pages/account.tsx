import PleaseSignIn from '../components/PleaseSignIn';

export default function Account() {
  return (
    <PleaseSignIn>
      <p>Account</p>
    </PleaseSignIn>
  );
}
