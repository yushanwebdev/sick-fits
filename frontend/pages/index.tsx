import { ALL_PRODUCTS_QUERY } from "../components/Products";
import { perPage } from "../config";
import { addApolloState, initializeApollo } from "../lib/apolloClient";
import { CURRENT_USER_QUERY } from "../lib/useUser";
import ProductsPage from "./products";

export async function getServerSideProps() {
  try {
    const apolloClient = initializeApollo();

    await apolloClient.query({
      query: ALL_PRODUCTS_QUERY,
      variables: {
        skip: 0,
        first: perPage,
      },
    });

    await apolloClient.query({
      query: CURRENT_USER_QUERY,
    });

    return addApolloState(apolloClient, {
      props: {},
    });
  } catch (error) {
    console.log(error);
  }
}

export default ProductsPage;
