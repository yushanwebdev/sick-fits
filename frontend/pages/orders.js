export default function Home() {
    return (
      <div>
        <p>Orders</p>
      </div>
    )
  }
  