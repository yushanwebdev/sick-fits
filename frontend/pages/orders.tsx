import { gql, useQuery } from "@apollo/client";
import Head from "next/head";
import Link from "next/link";
import styled from "styled-components";
import DisplayError from "../components/DisplayError";
import PleaseSignIn from "../components/PleaseSignIn";
import OrderItemStyles from "../components/styles/OrderItemStyles";
import formatMoney from "../lib/formatMoney";

const USER_ORDERS_QUERY = gql`
  query UserOrdersQuery {
    allOrders {
      id
      charge
      total
      user {
        id
      }
      items {
        id
        name
        description
        price
        quantity
        photo {
          image {
            publicUrlTransformed
          }
        }
      }
    }
  }
`;

const OrderUlStyles = styled.ul`
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(350px, 1fr));
  grid-gap: 4rem;
`;

function countItemsInAnOrder(order) {
  return order.items.reduce((tally, item) => tally + item.quantity, 0);
}

export default function Orders() {
  const { data, error, loading } = useQuery(USER_ORDERS_QUERY);

  if (loading) {
    return <p>Loading...</p>;
  }

  if (error) {
    return <DisplayError error={error} />;
  }

  const { allOrders } = data;

  return (
    <div>
      <Head>
        <title>Your Orders {allOrders.length}</title>
      </Head>
      <PleaseSignIn>
        <>
          <h2>You have {allOrders.length} orders!</h2>
          <OrderUlStyles>
            {allOrders.map((order) => (
              <OrderItemStyles key={order.id}>
                <Link href={`/order/${order.id}`}>
                  <div className="order-meta">
                    <p>
                      {countItemsInAnOrder(order)} Item
                      {countItemsInAnOrder(order) > 1 ? "s" : ""}
                    </p>
                    <p>
                      {order.items.length} Product
                      {order.items.length > 1 ? "s" : ""}
                    </p>
                    <p>{formatMoney(order.total)}</p>
                  </div>
                  <div className="images">
                    {order.items.map((item) => (
                      <img
                        key={item.id}
                        src={item.photo?.image?.publicUrlTransformed}
                        alt={item.name}
                      />
                    ))}
                  </div>
                </Link>
              </OrderItemStyles>
            ))}
          </OrderUlStyles>
        </>
      </PleaseSignIn>
    </div>
  );
}
