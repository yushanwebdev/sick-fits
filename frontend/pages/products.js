export default function Home() {
    return (
      <div>
        <p>Products</p>
      </div>
    )
  }
  