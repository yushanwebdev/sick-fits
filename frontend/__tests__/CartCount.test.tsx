import { act, render, screen, waitFor } from "@testing-library/react";
import { before } from "lodash";
import { ThemeProvider } from "styled-components";
import wait from "waait";
import CartCount from "../components/CartCount";
import { theme } from "../pages/_app";

beforeAll(() => {
  jest.useFakeTimers();
});

afterAll(() => {
  jest.useRealTimers();
});

describe("<CartCount />", () => {
  it("Renders", () => {
    render(
      <ThemeProvider theme={theme}>
        <CartCount count={10} />
      </ThemeProvider>
    );
  });

  it("Matches snapshot", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <CartCount count={10} />
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it("updates via props", async () => {
    const { container, rerender, debug } = render(
      <ThemeProvider theme={theme}>
        <CartCount count={11} />
      </ThemeProvider>
    );

    expect(container.textContent).toBe("11");
    // same
    // expect(container).toHaveTextContent("11");

    // Update the props
    rerender(
      <ThemeProvider theme={theme}>
        <CartCount count={12} />
      </ThemeProvider>
    );
    expect(container.textContent).toBe("1211");
    act(() => jest.advanceTimersByTime(400));
    expect(container.textContent).toBe("12");
    expect(container).toMatchSnapshot();
  });
});
