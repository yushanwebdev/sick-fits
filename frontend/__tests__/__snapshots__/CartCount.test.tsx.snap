// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount /> Matches snapshot 1`] = `
<div>
  <span
    class="sc-9b1afd6b-1 ZLHoM"
  >
    <div>
      <div
        class="sc-9b1afd6b-0 jEceeG count"
      >
        10
      </div>
    </div>
  </span>
</div>
`;

exports[`<CartCount /> updates via props 1`] = `
<div>
  <span
    class="sc-9b1afd6b-1 ZLHoM"
  >
    <div>
      <div
        class="sc-9b1afd6b-0 jEceeG count count-enter-done"
      >
        12
      </div>
    </div>
  </span>
</div>
`;
